.github/actions/setup-win/action.yml

```diff
@@ -37,13 +37,46 @@ runs:
       run: |
         Add-MpPreference -ExclusionPath $(Get-Location).tostring() -ErrorAction Ignore
 
+    - name: Setup conda
+      shell: bash
+      run: |
+        # Windows conda is baked into the AMI at this location
+        CONDA="C:\Jenkins\Miniconda3\condabin\conda.bat"
+
+        {
+          echo "CONDA_RUN=${CONDA} run --no-capture-output";
+          echo "CONDA_BUILD=${CONDA} run conda-build";
+          echo "CONDA_INSTALL=${CONDA} install";
+        } >> "${GITHUB_ENV}"
+
     - name: Setup Python3
-      uses: actions/setup-python@v4
-      with:
-        python-version: '3.x'
-        check-latest: false
-        cache: pip
-        cache-dependency-path: |
-          **/requirements.txt
-          **/.circleci/docker/requirements-ci.txt
-          **/.github/requirements-gha-cache.txt
+      shell: bash
+      run: |
+        set +e
+
+        PYTHON3=$(${CONDA_RUN} which python3)
+        HAS_PYTHON3=$?
+
+        if [[ "${HAS_PYTHON3}" == "0" ]]; then
+          echo "Found Python3 at ${PYTHON3}, adding it into GITHUB_PATH"
+          echo "${PYTHON3}" >> "${GITHUB_PATH}"
+        else
+          # According to https://docs.conda.io/en/latest/miniconda.html, we are using the Miniconda3
+          # installation, which is Python 3 based. Its Python is default to Python 3. Further, there
+          # is also the Miniconda installation that is Python 2 based, and both can be installed if
+          # needed.  In both cases, Python binary is just called python
+          PYTHON=$(${CONDA_RUN} which python)
+          HAS_PYTHON=$?
+
+          if [[ "${HAS_PYTHON}" == "0" ]]; then
+            PYTHON3=$(echo "${PYTHON}" | sed "s/python/python3/")
```

Comment:
There can be a python3 binary here that doesn't show up in the `conda which python3` command up top? That's surprising

Reply:
Ah no, this basically replace python with python3 as an alias name, i.e. `/c/jenkins/miniconda3/python` string would become /c/jenkins/miniconda3/python3` (Windows bash convention).  I then use the latter as the alias name, so future steps will be able to just call `python3`

Reply:
Maybe I'm misunderstanding something. `/c/jenkins/miniconda3/python3` represents a physical file on the computer, not an alias, right?

And in that case, shouldn't it have shown up when you ran `which python3` earlier?

Reply:
Yeah, it shouldn't show up.  If it's there, it would be use right away for `python3`.  If it's not there, it would be created by the `ln -s` command and points to `python`.